source/faq.txt

```diff
@@ -1,17 +1,103 @@
+.. _node-faq:
+
 ===
 FAQ
 ===
 
-.. default-domain:: mongodb
-
-Frequently Asked Questions
-
 .. contents:: On this page
    :local:
    :backlinks: none
-   :depth: 2
+   :depth: 1
    :class: singlecol
 
+This page contains frequently asked questions and their corresponding answers.
+
+.. tip::
+
+   If you can't find an answer to your problem on this page,
+   see the :ref:`node-issues-help` page for next steps and more
+   resources.
+
+.. _node-faq-connection-pool:
+
+How Does Connection Pooling Work in the Node Driver?
+----------------------------------------------------
+
+Every ``MongoClient`` instance has a built-in connection pool for each server
+in your MongoDB topology. Connection pools open sockets on demand to
+support concurrent requests to MongoDB in your application.
+
+The maximum size of each connection pool is set by the ``maxPoolSize`` option, which
+defaults to ``100``. If the number of in-use connections to a server reaches
+the value of ``maxPoolSize``, the next request to that server will wait
+until a connection becomes available.
+
+Each ``MongoClient`` instance opens two additional sockets per server in your
+MongoDB topology for monitoring the server's state.
+
+For example, a client connected to a 3-node replica set opens 6
+monitoring sockets. It also opens as many sockets as needed to support
+requests on each server, up to the value of ``maxPoolSize``. If
+``maxPoolSize`` is ``100`` and the application only uses the primary
+(the default), then only the primary connection pool grows and there can
+be at most ``106`` total connections. If the application uses a
+:manual:`read preference </core/read-preference/#read-preference>` to
+query the secondary nodes, their pools also grow and there can be
+``306`` total connections.
+
+Additionally, connection pools are rate-limited such that each connection pool
+can only create, at maximum, the value of ``maxConnecting`` connections
+in parallel at any time. Any additional request stops waiting in the
+following cases:
+
+- The connection pool finishes creating a connection and there are fewer
+  than ``minPoolSize`` connections in the pool.
+- An existing connection is checked back into the pool.
+- The driver's ability to reuse existing connections improves due to
+  rate-limits on connection creation.
+
+You can set the minimum number of concurrent connections to
+each server with the ``minPoolSize`` option, which defaults to ``0``.
+The connection pool will be initialized with this number of sockets. If
+sockets are closed due to any network errors, causing the total number
+of sockets (both in use and idle) to drop below the minimum, more
+sockets are opened until the minimum is reached.
+
+You can set the maximum number of milliseconds that a connection can
+remain idle in the pool before being removed and replaced with
+the ``maxIdleTimeMS`` option, which defaults to ``0`` (no limit).
```

Comment:
[suggestion]

Consider breaking this up. It exceeds the recommended sentence length and can be difficult to parse.

```suggestion
You can use ``maxIdleTimeMS`` to set the maximum number of milliseconds a connection can
remain idle in the pool. Once a connection has been idle for ``maxIdleTimeMS``, the driver removes and replaces it. ``maxIdleTimeMS`` defaults to ``0`` and has no upper limit.
```